include/net/ethernet.h

```diff
@@ -673,6 +673,17 @@ int net_eth_promisc_mode(struct net_if *iface, bool enable);
  */
 struct device *net_eth_get_ptp_clock(struct net_if *iface);
 
+/**
+ * @brief Return PTP clock that is tied to this ethernet network interface
+ * index.
+ *
+ * @param idx Network interface index
+ *
+ * @return Pointer to PTP clock if found, NULL if not found or if this
+ * ethernet interface index does not support PTP.
+ */
+struct device *net_eth_get_ptp_clock_by_index(int index);
```

Comment:
I prefer having an ifdef here, with a dummy function when not enabled. As it is done right below for net_eth_get_ptp_port()

Reply:
Sure, it can be done like that too.